### PR TITLE
 [ASDisplayNode] Fix an issue that causes a node to sometimes return an outdated calculated size or size range

### PR DESCRIPTION
**Pull request by nguyenhuy**
When asked for the calculated size or constrained size for the calculated layout, prefer the pending layout only if its version is up-to-date. This should fix the issue reported by @codeOfRobin ([here](https://gfycat.com/WastefulHideousDeviltasmanian)) and reproducible by [this sample project](https://github.com/codeOfRobin/TestTextureCell).

### CODE REVIEW

File: Source/ASDisplayNode+Layout.mm [hunk 1]
```diff
@@ -305,24 +305,22 @@ - (void)_u_measureNodeWithBoundsIfNecessary:(CGRect)bounds
   }
   
   CGSize boundsSizeForLayout = ASCeilSizeValues(bounds.size);
-  NSUInteger calculatedVersion = _calculatedDisplayNodeLayout->version;
```
Comment (nguyenhuy) [inline Source/ASDisplayNode+Layout.mm]: Note to reviewers: No change in the logic of this method (`-_u_measureNodeWithBoundsIfNecessary:`), just call the new `isValid()` function to (hopefully) increase readability.

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/13da3277458a9b7884cc221eb6acf947/raw/bce32295979a9ed71025a407a87e6e11601f3da7/log.txt)

**Comment by codeOfRobin** [comment]
@nguyenhuy any updates on this?

**Comment by codeOfRobin** [comment]
Wooohoo 🙌  🎉  ! I depended heavily on this feature working. Thank you so much guys 😄

**Comment by nguyenhuy** [comment]
@Adlai-Holler Thanks for syncing with master!
@codeOfRobin Thanks for reporting this bug!

**Pull request merged**